ncm-download/src/main/perl/download.pm

```diff
@@ -216,7 +217,9 @@ sub get_head
     local $ENV{'HTTPS_CA_FILE'} = $opts{cacert} if (exists($opts{cacert}));
     local $ENV{'HTTPS_CA_DIR'} = $opts{capath} if (exists($opts{capath}));
 
-    return LWP::UserAgent->new->head($source);
+    my $lwp = LWP::UserAgent->new;
+    $lwp->timeout($opts{head_timeout}) if (exists($opts{head_timeout}));
```

Comment:
if i read the code correct, `$opts{head_timeout}` will always exist, but could be undefined. so this should probably be `if (defined($opts{head_timeout}))`